osv/debian_version_cache.py

```diff
@@ -14,31 +14,29 @@
 """Caches the debian package first version dataframe"""
 import gzip
 import typing
-from urllib import request
+import requests
 import json
 
-CLOUD_API_CACHE_URL = 'https://storage.googleapis.com/debian-osv/first_package_cache.json.gz'
-debian_version_cache = None
+from . import cache
 
+CLOUD_API_CACHE_URL_TEMPLATE = 'https://storage.googleapis.com/debian-osv/first_package_output/{version}/{package}'
+CACHE_DURATION_SECONDS = 60 * 60 * 24
 
-def _initiate_from_cloud_cache(
-    force_reload: bool = False) -> typing.Dict[str, typing.Dict]:
-  """Load the debian version cache from the Google cloud link, if it hasn't been loaded yet"""
-  global debian_version_cache
-  if debian_version_cache is None or force_reload:
-    res = request.urlopen(CLOUD_API_CACHE_URL)
-    debian_version_cache = json.loads(gzip.decompress(res.read()))
+debian_version_cache = cache.InMemoryCache()
 
 
+@cache.Cached(
+    debian_version_cache, 24 * 60 * 60, cache_blacklist=frozenset({'0'}))
```

Comment:
Also, we're not using `cache_blacklist` right? Would prefer to remove it add and it later if we ever need it (YAGNI).

Reply:
Well this is using it, when the web returns 404, we don't want to cache '0' response right? since it means it's a new package so that information will be outdated very soon. 

I suppose 24 hours cache is not very long, especially since the first_package_finder only runs once every 24 hours anyway, so it can be removed.

Reply:
Oops I somehow missed that. My bad! 

We should just move this logic to the caller instead, and raise an exception in this case to make this behaviour more transparent.
i.e. in `ecosystems.py`

```python
try:
     introduced = debian_version_cache.get_first_package_version(                                                                                                                                                                                                                                                                   
          package, self.debian_release_ver) 
except VersionNotExist:
    pass  # introduced is already '0' at this point
```